lib/src/modules/conditional.dart

```diff
@@ -46,19 +49,22 @@ abstract class _Always extends Module with CustomSystemVerilog {
         If(
           reset,
           then: [
-            ...allReceivers.map((rec) =>
-                rec < (((resetVal?[rec]) != null) ? (resetVal?[rec]) : 0))
-            // If resetVal for a receiver is defined,
-            // then use it for assigning receiver
-            // else assign zero as resetVal
+            ...allReceivers.map((rec) {
+              final driver = resetValues?[rec] ?? 0;
+              return rec < driver;
+              // If resetValue for a receiver is defined,
+              // then use it for assigning receiver
+              // else assign zero as resetValue
+            })
           ],
           orElse: conditionals,
         ),
       ];
-    } else {
-      conditionals = conditionals;
     }
 
+    // ignore: prefer_initializing_formals
+    this.conditionals = conditionals;
```

Comment:
If `List<Conditional> conditionals` in the constructor is replaced with `this.conditionals`, then this is no longer needed.